src/server.lisp

```diff
@@ -158,90 +158,99 @@ DISPATCH-TABLE and LOGGING-STREAM are both required arguments.  TIMEOUT is of ty
-                        (let ((f (gethash (|RPCRequest-method| request) dispatch-table)))
-                          (unless f
-                            (error 'unknown-rpc-method
-                                   :method-name (|RPCRequest-method| request)))
-                          (setf result 
-                                (if timeout
-                                    (bt:with-timeout (timeout)
-                                      (apply f (append args-as-list kwargs-as-plist)))
-                                    (apply f (append args-as-list kwargs-as-plist))))
-                          
-                          (setf reply (make-instance '|RPCReply|
-                                                     :|id| (|RPCRequest-id| request)
-                                                     :|result| result))))
-                      (log-completion-message :info "Requested ~a completed" (|RPCRequest-method| request)))
-                  
-                  ;; this is where errors go where we can reply to the client
-                  (unknown-rpc-method (c)
-                    (declare (ignore c))
-                    (log-completion-message :err "Request ~a error: method ~a unknown"
-                                            (|RPCRequest-id| request)
-                                            (|RPCRequest-method| request))
-                    (setf reply (make-instance '|RPCError|
-                                               :|id| (|RPCRequest-id| request)
-                                               :|error| (format nil "Method named \"~a\" is unknown."
-                                                                (|RPCRequest-method| request)))))
-                  (bt:timeout (c)
-                    (declare (ignore c))
-                    (log-completion-message :err "Request ~a error: timed out"
-                                            (|RPCRequest-id| request))
-                    (setf reply (make-instance '|RPCError|
-                                               :|id| (|RPCRequest-id| request)
-                                               :|error| (format nil "Execution timed out.  Note: time limit: ~a seconds." timeout))))
-                  (error (c)
-                    (log-completion-message :err
-                                            "Request ~a error: Unhandled error in host program:~%~a"
-                                            (|RPCRequest-id| request)
-                                            c)
-                    (setf reply (make-instance '|RPCError|
-                                               :|id| (|RPCRequest-id| request)
-                                               :|error| (format nil "~a" c)))))
-                
-                ;; send the client response, whether success or failure
-                (%push-raw-request receiver identity empty-frame (serialize reply)))))
+       (handler-case
```

Comment:
Emacs reindented this for me, but the real change is the `handler-bind` and the `:|warnings|` slot initializers.